src/Microsoft.OData.Core/Json/ODataUtf8JsonWriter.TextWriter.cs

```diff
@@ -0,0 +1,489 @@
+﻿//---------------------------------------------------------------------
+// <copyright file="ODataUtf8JsonWriter.TextWriter.cs" company="Microsoft">
+//      Copyright (C) Microsoft Corporation. All rights reserved. See License.txt in the project root for license information.
+// </copyright>
+//---------------------------------------------------------------------
+
+#if NETCOREAPP
+namespace Microsoft.OData.Json
+{
+    using System;
+    using System.Buffers;
+    using System.Diagnostics;
+    using System.IO;
+    using System.Text;
+    using System.Text.Unicode;
+    using System.Threading.Tasks;
+
+    internal sealed partial class ODataUtf8JsonWriter
+    {
+        /// <summary>
+        /// Starts a TextWriter value scope with the specified content type.
+        /// </summary>
+        /// <param name="contentType">The content type of the TextWriter value scope.</param>
+        /// <returns>A TextWriter for writing JSON data.</returns>
+        public TextWriter StartTextWriterValueScope(string contentType)
+        {
+            this.CommitUtf8JsonWriterContentsToBuffer();
+
+            WriteItemWithSeparatorIfNeeded();
+
+            this.currentTextWriterContentType = contentType;
+            bool isWritingJson = String.IsNullOrEmpty(this.currentTextWriterContentType) || this.currentTextWriterContentType.StartsWith(MimeConstants.MimeApplicationJson, StringComparison.Ordinal);

```

Comment:
nit:
```suggestion
            bool isWritingJson = string.IsNullOrEmpty(this.currentTextWriterContentType) || this.currentTextWriterContentType.StartsWith(MimeConstants.MimeApplicationJson, StringComparison.Ordinal);
```
It's very likely the analyzer is flagging this...